### PR TITLE
Move locks to /run/storage/locks

### PR DESCRIPTION
**Pull request by rhatdan**
This could increase spead, and should allow us to have read only image
and layer stores

Signed-off-by: Dan Walsh <dwalsh@redhat.com>

### CODE REVIEW

**Comment by rhatdan** [comment]
@nalind PTAL

**Comment by rhatdan** [comment]
@rhvgoyal PTAL

**Comment by nalind** [comment]
This creates breakage if a shared location is going to be used for read-write from more than one location.

File: storage/containers.go [hunk 1]
```diff
@@ -170,7 +170,7 @@ func newContainerStore(dir string) (ContainerStore, error) {
 	if err := os.MkdirAll(dir, 0700); err != nil {
 		return nil, err
 	}
-	lockfile, err := GetLockfile(filepath.Join(dir, "containers.lock"))
+	lockfile, err := GetLockfile(filepath.Join(lockDir, dir, "containers.lock"))
```
Comment (nalind) [inline storage/containers.go]: `dir` is already an absolute path, so this creates a lock file named `/run/storage/locks/var/lib/containers/storage/containers/containers.lock`.  It looks weird, but it's consistent and unique among different graph roots, so I guess it's fine.

**Comment by rhatdan** [comment]
> This creates breakage if a shared location is going to be used for read-write from more than one location.

Please explain?

**Comment by rhvgoyal** [comment]
So you are moving all locks to a tmpfs (/run) ?

**Comment by rhatdan** [comment]
Yes the main reason for this is so that alternate container/storage repos can be on read/only file systems. (NFS for example).  I would figure the locking would also be quicker on a tmpfs then on physical disk, especially if the disk is remote.  Only issue I am aware of would be locking form other systems, but I don't think we could do a good job with that anyways. 
Potentially someone could remove some nfs data while a tool on a host is using it, but this could happen in thousands of different ways.  We are only trying to handle local locking.  We need to tell users that they would need to be careful with updating container stores if they are in use.

**Comment by rhatdan** [comment]
@nalind I now made this configurable so if you want to have locks on the nfs shares it will work.

**Comment by rhatdan** [comment]
If we merge https://github.com/containers/storage/pull/52 we can close this one.

**Pull request closed without merging**